frontend/src/components/dashboard/aliases/AliasList.tsx

```diff
@@ -0,0 +1,151 @@
+};
+
+/**
+ * Display a list of <Alias> cards, with the ability to filter them or create a new alias.
+ */
+export const AliasList = (props: Props) => {
+  const { l10n } = useLocalization();
+  const [stringFilterInput, setStringFilterInput] = useState("");
+  const [categoryFilters, setCategoryFilters] = useState<SelectedFilters>({});
+  const [localLabels, storeLocalLabel] = useLocalLabels();
+
+  if (props.aliases.length === 0) {
+    return null;
+  }
+
+  const aliases = sortAliases(
+    filterAliases(props.aliases, props.profile, {
+      ...categoryFilters,
+      string: stringFilterInput,
+    })
+  );
+
+  const aliasCards = aliases.map((alias, index) => {
+    const aliasWithLocalLabel = { ...alias };
+    if (
+      alias.description.length === 0 &&
+      props.profile.server_storage === false &&
+      localLabels !== null
+    ) {
+      const type = isRandomAlias(alias) ? "random" : "custom";
+      const localLabel = localLabels.find(
+        (localLabel) => localLabel.id === alias.id && localLabel.type === type
+      );
+      if (localLabel !== undefined) {
+        aliasWithLocalLabel.description = localLabel.description;
+      }
+    }
+
+    const onUpdate = (updatedFields: Partial<AliasData>) => {
+      if (
+        localLabels !== null &&
+        typeof updatedFields.description === "string"
+      ) {
+        storeLocalLabel(alias, updatedFields.description);
+      }
+      return props.onUpdate(alias, updatedFields);
+    };
+
+    return (
+      <li
+        className={styles.aliasCardWrapper}
+        key={alias.address + isRandomAlias(alias)}
+      >
+        <Alias
+          alias={aliasWithLocalLabel}
+          user={props.user}
+          profile={props.profile}
+          onUpdate={onUpdate}
+          onDelete={() => props.onDelete(alias)}
+          defaultOpen={index === 0}
+          showLabelEditor={props.profile.server_storage || localLabels !== null}
+        />
+      </li>
+    );
+  });
+
+  // With at most five aliases, filters aren't really useful
+  // for non-Premium users.
+  const filters = props.profile.has_premium ? (
+    <>
+      <div className={styles.stringFilter}>
+        <VisuallyHidden>
+          <label htmlFor="stringFilter">
+            {l10n.getString("profile-filter-search-placeholder")}
+          </label>
+        </VisuallyHidden>
+        <input
+          value={stringFilterInput}
+          onChange={(e) => setStringFilterInput(e.target.value)}
+          type="search"
+          name="stringFilter"
+          id="stringFilter"
+          placeholder={l10n.getString("profile-filter-search-placeholder")}
+        />
+        <span className={styles.matchCount}>
+          {aliases.length}/{props.aliases.length}
+        </span>
+      </div>
+      <div className={styles.categoryFilter}>
+        <CategoryFilter
+          onChange={setCategoryFilters}
+          selectedFilters={categoryFilters}
+        />
+      </div>
+    </>
+  ) : null;
```

Comment:
Yeah agreed, I had to take some design liberties there since the current website doesn't have this either, but: 6a44caf86d0ff1459c0d2f850a042e671a366765.

Reply:
Thats very helpful — thanks for adding this!